workers/unity/Packages/com.improbable.gdk.core/GameObjectRepresentation/Injection/WorldCommandResponseHandler.cs

```diff
@@ -0,0 +1,159 @@
+            public Commands.WorldCommands.DeleteEntity.Request RequestPayload { get; }
+            public readonly object Context;
+
+            public DeleteEntityResponse(Commands.WorldCommands.DeleteEntity.ReceivedResponse response, object context)
+            {
+                RequestPayload = response.RequestPayload;
+                Op = response.Op;
+                Context = context;
+            }
+        }
+
+        [InjectableId(InjectableType.WorldCommandResponseHandler)]
+        [InjectionCondition(InjectionCondition.RequireNothing)]
+        public class WorldCommandResponseHandler : IInjectable
+        {
+            private readonly Entity entity;
+            private readonly ILogDispatcher logDispatcher;
+
+            private readonly List<Action<CreateEntityResponse>> createEntityDelegates
+                = new List<Action<CreateEntityResponse>>();
+
+            private readonly List<Action<ReserveEntityIdsResponse>> reserveEntityIdsDelegates
+                = new List<Action<ReserveEntityIdsResponse>>();
+
+            private readonly List<Action<DeleteEntityResponse>> deleteEntityDelegates
+                = new List<Action<DeleteEntityResponse>>();
+
+            public event Action<ReserveEntityIdsResponse> OnReserveEntityIdsResponse
+            {
+                add => reserveEntityIdsDelegates.Add(value);
+                remove => reserveEntityIdsDelegates.Remove(value);
+            }
+
+            public event Action<CreateEntityResponse> OnCreateEntityResponse
+            {
+                add => createEntityDelegates.Add(value);
+                remove => createEntityDelegates.Remove(value);
+            }
+
+            public event Action<DeleteEntityResponse> OnDeleteEntityResponse
+            {
+                add => deleteEntityDelegates.Add(value);
+                remove => deleteEntityDelegates.Remove(value);
+            }
+
+            private WorldCommandResponseHandler(Entity entity,
+                World world,
+                ILogDispatcher logDispatcher)
+            {
+                world.GetOrCreateManager<GameObjectWorldCommandSystem>().RegisterResponseHandler(entity, this);
+
+                this.entity = entity;
+                this.logDispatcher = logDispatcher;
+            }
+
+            internal void OnCreateEntityResponseInternal(
+                Commands.WorldCommands.CreateEntity.ReceivedResponse receivedResponse)
+            {
+                if (receivedResponse.Context is WorldCommandContext worldCommandContext
+                    && worldCommandContext.OwnerEntity == entity)
+                {
+                    GameObjectDelegates.DispatchWithErrorHandling(
+                        new CreateEntityResponse(receivedResponse, worldCommandContext.Context),
+                        createEntityDelegates,
+                        logDispatcher);
+                }
+            }
+
+            internal void OnReserveEntityIdsResponseInternal(
```

Comment:
Super nit: Can we swap the order here to have `OnReserveEntityIdsResponseInternal` before `OnCreateEntityResponseInternal`?